.github/workflows/ci.yml

```diff
@@ -102,3 +102,10 @@ jobs:
                   api-level: ${{ matrix.api-level }}
                   arch: x86
                   script: ./gradlew connectedCheck -PuseCompose=${{ matrix.compose-enabled }}
+
+            - name: Upload test reports
+              if: always()
```

Comment:
Maybe `if: failed()` is better?

Reply:
Also might be useful to add the same upload step to other steps like detekt and unit tests.

Reply:
great points. will make these changes =]